pinot-spi/src/main/java/org/apache/pinot/spi/trace/ServerStatsInfo.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.trace;
```

Comment:
If we are putting this in the SPI we have to take care of a few things.

1. A Public ctor with AllArgs is not going to be tenable, since in the future we may want to add more properties to this. Suggest using NoArgs ctor with setters that return `this`, or builder pattern. We can also check what convention Pinot uses right now.
2. We should define precisely each of the fields here. You can add a javadoc for each field member var.
3. Could we get rid of redundancies like `ServerStats` in `SingleConnectionBrokerRequestHandler`? If this is in the SPI would make sense to make this the broadly used class for handling this info.